### ISSUE TITLE
Only name and id returned regardless of info_fields

### ISSUE THREAD

**Issue by crystalneth** [simi/omniauth-facebook#236]
I have tried setting info_fields to any number of things, including just 'first_name', but only name and id are returned from facebook.

I don't see anywhere in omniauth source code where info_fields are passed to facebook as the fields parameter, which would be required for facebook to return those fields. The request to /me in the library only requests the default fields.

My workaround is to request the fields myself.

**Comment by rlugge** [comment]
I'm getting the exact same issue.  How are you requesting the additional pieces of information, @crystalneth ?

**Comment by Argonus** [comment]
@crystalneth Have you resolved this problem? Facebook give me back only email/full name and id...

**Comment by mkdynamic** [comment]
What are you passing for `scope` -- to include permissions for accessing this info? By default, only `email` is requested.

**Comment by ssborges** [comment]
Hi @crystalneth, I have the same problem. I am not able to fetch any other field (also `Extra` only returns Id and Name), despite my efforts to change the `scope` to include permissions. Have you resolved this problem?  thx

**Comment by shadowed87** [comment]
I have been using this gem for a year, suddenly, no changes, updated to last version, and I am only getting email and fullname, regardless of any info_fields....updating does not solve the issue for me. Any hints?

**Comment by mkdynamic** [comment]
@shadowed87 What scope are you passing?

**Comment by shadowed87** [comment]
Ok, after many tests, it's going through, passing the right configuration. There was a problem with the interaction with devise, written a while ago but for some reason now there was a missing comma XD (after ...APP_SECRET']), which wasn't throwing any error due to the brackets.

`
config.omniauth` :facebook, ENV['FACEBOOK_APP_ID'], ENV['FACEBOOK_APP_SECRET']
{
   :image_size => {:width => 165, :height => 165},
   :scope => "email, public_profile, publish_actions",
   :info_fields => 'id, email, first_name, last_name, link, picture'
}
`

Now I have anyway updated it, and everything is passing through

`
config.omniauth :facebook, ENV['FACEBOOK_APP_ID'], ENV['FACEBOOK_APP_SECRET'],
            scope: 'public_profile, publish_actions, user_about_me',
            info_fields: 'id, email, first_name, last_name, link, picture',
            image_size: {:width => 165, :height => 165}
`

anyway, sorry for the false positive :P

**Comment by mkdynamic** [comment]
@shadowed87 :thumbsup: thank for sharing.

**Comment by nasirshamshadsynnapps** [comment]
I'm facing the same issue, only ID and NAME coming through...
Here is my devise.rb configuration...
 
config.omniauth :facebook, 'APP_ID', 'APP_SEC', scope: 'public_profile, publish_actions, user_about_me', info_fields: 'email,id,name,first_name,last_name,age_range,link,gender,timezone,updated_time,verified', callback_url: "http://localhost:3000/users/auth/facebook/callback", secure_image_url: true, image_size: 'large'

**Comment by tjankov** [comment]
Has this been solved?

**Comment by nasirshamshadsynnapps** [comment]
No yet... @tjankov

**Comment by pduersteler** [comment]
Just had the same problem after it worked for some time and found a solution, at least in my case. Maybe it helps someone.

I was testing an app through my sandbox/test app. Inside the [facebook graph api explorer](https://developers.facebook.com/tools/explorer/), I selected "User token", added the "email" permission (again, as I removed it for testing something else a while ago). Requesting a new token through the api explorer and giving the permission, I had the email address again in the oauth response.
